core/service/src/client/tests/threshold/mpc_context_tests.rs

```diff
@@ -0,0 +1,204 @@
+        for node in new_context.kms_nodes.iter_mut() {
+            let pk: PublicSigKey = read_versioned_at_request_id(
+                &all_public_storage[node.party_id as usize - 1],
+                &SIGNING_KEY_ID,
+                &PubDataType::VerfKey.to_string(),
+            )
+            .await
+            .unwrap();
+            node.verification_key = Some(pk);
+        }
+        new_context
+    };
+    let new_context_id = *new_context.context_id();
+    println!("got new context: {:?}", new_context);
+
+    {
+        let req = internal_client
+            .new_kms_context_request(Some(&previous_context_id), new_context)
+            .unwrap();
+
+        let mut req_tasks = JoinSet::new();
+        for (_, client) in kms_clients.iter() {
+            let req_clone = req.clone();
+            let mut client = client.clone();
+            req_tasks.spawn(async move { client.new_kms_context(req_clone).await });
+        }
+
+        let mut req_response_vec = Vec::new();
+        while let Some(inner) = req_tasks.join_next().await {
+            req_response_vec.push(inner.unwrap().unwrap().into_inner());
+        }
+        assert_eq!(req_response_vec.len(), kms_clients.len());
+    }
+
+    // run a decryption test
+    let enc_config = EncryptionConfig {
+        compression: true,
+        precompute_sns: true,
+    };
+    let key_id = &TEST_THRESHOLD_KEY_ID_4P;
+    run_decryption_threshold(
+        amount_parties,
+        &mut kms_servers,
+        &mut kms_clients,
+        &mut internal_client,
+        key_id,
+        Some(&new_context_id),
+        vec![TestingPlaintext::Bool(true)],
+        enc_config,
+        None,
+        1,
+        None,
+    )
+    .await;
+
+    // delete the new context
+    {
+        let req = internal_client
+            .destroy_kms_context_request(&new_context_id)
+            .unwrap();
+
+        let mut req_tasks = JoinSet::new();
+        for (_, client) in kms_clients.iter() {
+            let req_clone = req.clone();
+            let mut client = client.clone();
+            req_tasks.spawn(async move { client.destroy_kms_context(req_clone).await });
+        }
+
+        let mut req_response_vec = Vec::new();
+        while let Some(inner) = req_tasks.join_next().await {
+            req_response_vec.push(inner.unwrap().unwrap().into_inner());
+        }
+        assert_eq!(req_response_vec.len(), kms_clients.len());
+    }
+
+    // run the request again with the new context ID (which is deleted)
+    // this should fail.
+    run_decryption_threshold_optionally_fail(
+        amount_parties,
+        &mut kms_servers,
+        &mut kms_clients,
+        &mut internal_client,
+        key_id,
+        Some(&new_context_id),
+        vec![TestingPlaintext::Bool(false); 2],
+        enc_config,
+        None,
+        1,
+        None,
+        true,
+    )
+    .await;
+
+    // running the request with the old context ID should still work
```

Comment:
I guess the old context is the default one?
I think that eventually we don't really want a default context, but rather need to get an initial context from the gateway before we can do anything.
Of course this should be a follow up PR as it is a huge change :)